core/src/main/java/org/infinispan/remoting/inboundhandler/DefaultTopologyRunnable.java

```diff
@@ -39,19 +38,31 @@ public boolean isReady() {
    }
 
    @Override
-   protected Response beforeInvoke() throws Exception {
+   protected InvocationStatus beforeInvoke() throws Exception {
+      // Read commands are executed directly in OOB thread pool. When the many commands are received prior
+      // to the first topology update (when the node is joining), these can wait for transaction data
+      // (even in non-tx mode) and deplete OOB TP => the node stops being responsive event to ST.
```

Comment:
Can you change this comment to talk instead about commands that implement `TopologyAffectedCommand` and yet return `false` from `canBlock()`? Which, like I said before, I think is a bug in those commands ;)

BTW, waiting for tx data in non-tx mode means just waiting for the topology. Can you add a javadoc to `StateTransferLock.transactionDataFuture()` to mention that, instead of the comment here?